File: 404-sum-of-left-leaves/sum-of-left-leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(NULLptr), right(NULLptr) {}
 *     TreeNode(int x) : val(x), left(NULLptr), right(NULLptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sum=0;

    void sumOfLeavesHelper(TreeNode* node, bool left){
        if(node->left==NULL && node->right==NULL){
            if(left)sum+=node->val;
            return;
        }
        if(node->left!=NULL){
            sumOfLeavesHelper(node->left,true);
        }
        if(node->right!=NULL){
            sumOfLeavesHelper(node->right,false);
        }
    }
    int sumOfLeftLeaves(TreeNode* root) {
       if(root==NULL)return 0;
       
       if(root->left!=NULL){
            sumOfLeavesHelper(root->left,true);
        }
        if(root->right!=NULL){
            sumOfLeavesHelper(root->right,false);
        }

        return sum;
    }
};

File: 404-sum-of-left-leaves/sum-of-left-leaves.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function helper(node:TreeNode|null, left:boolean, sum:number): number{
    if(node.left===null && node.right===null){
        if(left)return node.val;
        else return 0;
    }
    
    let leftSum=0,rightSum=0;
    if(node.left!==null){
        leftSum=helper(node.left,true,sum);
    }

    if(node.right!==null){
        rightSum=helper(node.right,false,sum);
    }

    return leftSum+rightSum;

   
}
function sumOfLeftLeaves(root: TreeNode | null): number {
    if(root===null)return 0;

    return helper(root,false,0);

};